src/lib/update-lock.ts

```diff
@@ -70,95 +71,126 @@ export const readLock: LockFn = Bluebird.promisify(locker.async.readLock, {
 	context: locker,
 });
 
-// Unlock all lockfiles, optionally of an appId | appUuid, then release resources.
-function dispose(
-	release: () => void,
-	appIdentifier: string | number,
-): Bluebird<void> {
-	return Bluebird.map(
-		lockfile.getLocksTaken((p: string) =>
-			p.includes(`${BASE_LOCK_DIR}/${appIdentifier}`),
-		),
-		(lockName) => {
-			return lockfile.unlock(lockName);
-		},
-	)
-		.finally(release)
-		.return();
+async function dispose(appIdentifier: string | number): Promise<void> {
+	const locks = lockfile.getLocksTaken((p: string) =>
+		p.includes(`${BASE_LOCK_DIR}/${appIdentifier}`),
+	);
+	await Promise.all(locks.map((l) => lockfile.unlock(l)));
+}
+
+/**
+ * Attempts to take locks for list of application IDs provided
+ * Waits until all applications are locked before running the passed function
+ */
+export async function lockAll<T extends unknown>(
+	appIds: number[],
+	force = false,
+	fn: () => Resolvable<T>,
+): Promise<T | undefined> {
+	if (appIds.length === 0) {
+		throw new UpdatesLockedError(
```

Comment:
This should be an assertion or an `InternalInconsistencyError` as `UpdatesLockError` is caught upper in the call stack to report a service being locked.

Reply:
good one